### DISCUSSION TITLE
Layout enhancement

### DISCUSSION THREAD

**Discussion by kangz12345** [snu-quiqcl/qiwis#57] [Ideas]
The current layout of `swift` is quite not pretty. Like the dashboard of ARTIQ, we need some kind of spacers, which hold some space in the main window. This will prevent the frames from expanding to fill the main window.

**Comment by kangz12345** [comment]
@Ahrrri please discuss this further.

**Reply by Ahrrri** [reply]
Oh, I forgot about this 😮

**Reply by Ahrrri** [reply]
The dashboard of ARTIQ seems to use QDockWidgets, basically, in each tap.
Should we retain the QDockWidgets configuration, but just put some spaces between them?
![image](https://user-images.githubusercontent.com/75469641/224711491-852dba3d-6220-400a-b198-12d94f00a1a1.png)

**Reply by kangz12345** [reply]
I think we should decide this independently to ARTIQ! Could you compare the two?

**Reply by Ahrrri** [reply]
In a layout using several `QDockWidget`s, we need a central widget(by `setCentralWidget`), because `QDockWidget`s are "docked" to the central widget from four directions. Hence if a large center widget is needed then we may consider using `QDockWidget`, otherwise mdi layout would be better.

**Reply by Ahrrri** [reply]
Should we use one tap per each App? Then each tap has (QCentralWidget+some QDockWidgets) or (QMdiSubWindows in QMdiArea) layout.

**Reply by kangz12345** [reply]
> In a layout using several `QDockWidget`s, we need a central widget(by `setCentralWidget`), because `QDockWidget`s are "docked" to the central widget from four directions. Hence if a large center widget is needed then we may consider using `QDockWidget`, otherwise mdi layout would be better.

What a great observation!
